goose/recipes/update-centos-package.yaml

```diff
@@ -48,34 +61,48 @@ parameters:
-    * If there is no new version available, print a message and exit.  If there is a new version available follow the instructions below.
-    * Do not clone any repository for detecting the versions.
+  2. Check if the {{ package }} was not already updated to version {{ version }}.  That means comparing the current version and provided version.
+    * The current version of the package can be found in the 'Version' field of the RPM .spec file.
+    * If there is nothing to update, print a message and exit. Otherwise follow the instructions below.
+    * Do not clone any repository for detecting the version in .spec file.
 
   3. Create a local Git repository by following these steps:
     * Create a fork of the {{ package }} package using the glab tool.
     * Clone the fork into the temp directory.
-
+  
   4. Update the {{ package }} to the newer version:
-    * Create a new branch indicating the update of the package and that it was created by {{ git_user }}.
-    * Update the local package.
-    * Only perform changes relevant for the version update.
-    * Use the rpmlint tool to make sure that changes to the .spec file did not introduce new errors.
-    * Build the local .spec file to make sure all changes are correct.
-    * Do not rename variables or comment existing lines out.
-    * Make sure that the commit looks similar to previous ones in the Git log that update the package version.
-    * Do not change if-else branches.
+    * Create a new Git branch named `automated-packaging-update-{{ version }}`.
+    * Update the local package by:
+      * Updating the 'Version' and 'Release' fields in the .spec file as needed, following packaging documentation.
+      * Updating macros related to update (e.g., 'commit') if present and necessary; examine the file's history to see how updates are typically done.
+      * Creating a changelog entry, referencing the Jira issue as "Resolves {{ jira_issue }}".
+      * Downloading sources using `spectool -g -S {{ package }}.spec` and uploading the new sources using `centpkg --release {{ git_branch }} new-sources`.
+      * Only performing changes relevant to the version update: Do not rename variables, comment out existing lines, or alter if-else branches in the .spec file.
+      
+  5. Verify and adjust the changes:
+    * Use `rpmlint` to validate your .spec file changes and fix any new errors it identifies.
+    * Try to successfully build and test local .spec file and source changes. Follow these steps, fixing any errors along the way:
+      * Generate the SRPM using `rpmbuild -bs` (ensure your .spec file and source files are correctly copied to the build environment as required by rpmbuild).
+      * Perform a scratch build using the generated SRPM using `centpkg scratch-build --srpm <path-to-your-srpm>`
```

Comment:
Running just `centpkg scratch-build --srpm` should do the step above implicitly.

Reply:
for now kept it as separate steps for easier debugging